### ISSUE TITLE
Random Crash on start MapBox: Mapbox was compiled with optimization - stepping may behave oddly; variables may not be available.

### ISSUE THREAD

**Issue by pingus-nikalex** [mapbox/mapbox-gl-native#16265]
Steps to reproduce

Start Application with MapBox View
Actual behavior

Application crashed with message
Mapbox was compiled with optimization - stepping may behave oddly; variables may not be available.

Configuration

**Mapbox SDK versions:

Mapbox-iOS-SDK (5.5.1)
MapboxCoreNavigation (0.39.0):
MapboxDirections.swift (~> 0.30.0)
MapboxMobileEvents (~> 0.9.5)
MapboxNavigationNative (~> 6.2.1)
Turf (~> 0.3.0)
MapboxDirections.swift (0.30.0):
Polyline (~> 4.2)
MapboxMobileEvents (0.9.5)
MapboxNavigation (0.39.0):
Mapbox-iOS-SDK (= 5.5.1)
MapboxCoreNavigation (= 0.39.0)
MapboxSpeech (~> 0.1.0)
Solar (~> 2.1)
MapboxNavigationNative (6.2.1)
MapboxSpeech (0.1.1)
**iOS/macOS versions:
13.3.1
**Device/simulator models:
iPhone XS Max
**Xcode version:
11.3.1

Crashed thread:
com.mapbox.mbgl.DefaultFileSource (10)

Stack:
```
#0	0x0000000104c0b058 in mbgl::Mailbox::push(std::__1::unique_ptr<mbgl::Message, std::__1::default_deletembgl::Message >) at /Users/distiller/project/vendor/mapbox-gl-native/src/mbgl/actor/mailbox.cpp:60
#1	0x0000000104be4564 in void mbgl::ActorRefmbgl::FileSourceRequest::invoke<void (mbgl::FileSourceRequest::)(mbgl::Response const&), mbgl::Response const&>(void (mbgl::FileSourceRequest::)(mbgl::Response const&), mbgl::Response const&) const at /Users/distiller/project/vendor/mapbox-gl-native/include/mbgl/actor/actor_ref.hpp:34
#2	0x0000000104be185c in mbgl::DefaultFileSource::Impl::request(mbgl::AsyncRequest*, mbgl::Resource, mbgl::ActorRefmbgl::FileSourceRequest)::'lambda'(mbgl::Response const&)::operator()(mbgl::Response const&) const [inlined] at /Users/distiller/project/vendor/mapbox-gl-native/platform/default/src/mbgl/storage/default_file_source.cpp:108
#3	0x0000000104be1848 in mbgl::DefaultFileSource::Impl::request(mbgl::AsyncRequest*, mbgl::Resource, mbgl::ActorRefmbgl::FileSourceRequest) at /Users/distiller/project/vendor/mapbox-gl-native/platform/default/src/mbgl/storage/default_file_source.cpp:147
#4	0x0000000104be7b94 in void mbgl::MessageImpl<mbgl::DefaultFileSource::Impl, void (mbgl::DefaultFileSource::Impl::)(mbgl::AsyncRequest, mbgl::Resource, mbgl::ActorRefmbgl::FileSourceRequest), std::__1::tuple<mbgl::FileSourceRequest*, mbgl::Resource, mbgl::ActorRefmbgl::FileSourceRequest > >::invoke<0ul, 1ul, 2ul>(std::__1::integer_sequence<unsigned long, 0ul, 1ul, 2ul>) at /Users/distiller/project/vendor/mapbox-gl-native/include/mbgl/actor/message.hpp:34
#5	0x0000000104c0b250 in mbgl::Mailbox::receive() at /Users/distiller/project/vendor/mapbox-gl-native/src/mbgl/actor/mailbox.cpp:84
#6	0x0000000104c0bd60 in mbgl::Mailbox::maybeReceive(std::__1::weak_ptrmbgl::Mailbox) [inlined] at /Users/distiller/project/vendor/mapbox-gl-native/src/mbgl/actor/mailbox.cpp:94
#7	0x0000000104c0bd44 in mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0::operator()() const [inlined] at /Users/distiller/project/vendor/mapbox-gl-native/src/mbgl/actor/mailbox.cpp:100
#8	0x0000000104c0bd24 in decltype(std::__1::forward<mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0&>(fp)()) std::__1::__invoke<mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0&>(mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0&) [inlined] at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:4361
#9	0x0000000104c0bd24 in void std::__1::__invoke_void_return_wrapper::__call<mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0&>(mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0&) [inlined] at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__functional_base:349
#10	0x0000000104c0bd24 in std::__1::__function::__alloc_func<mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0, std::__1::allocator<mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0>, void ()>::operator()() [inlined] at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:1527
#11	0x0000000104c0bd24 in std::__1::__function::__func<mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0, std::__1::allocator<mbgl::Mailbox::makeClosure(std::__1::weak_ptrmbgl::Mailbox)::$_0>, void ()>::operator()() at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:1651
#12	0x0000000104bd9364 in std::__1::__function::__value_func<void ()>::operator()() const [inlined] at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:1799
#13	0x0000000104bd9350 in std::__1::function<void ()>::operator()() const [inlined] at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:2347
#14	0x0000000104bd9350 in void mbgl::WorkTaskImpl<std::__1::function<void ()>, std::__1::tuple<> >::invoke<>(std::__1::integer_sequence) [inlined] at /Users/distiller/project/vendor/mapbox-gl-native/include/mbgl/util/work_task_impl.hpp:43
#15	0x0000000104bd9350 in mbgl::WorkTaskImpl<std::__1::function<void ()>, std::__1::tuple<> >::operator()() at /Users/distiller/project/vendor/mapbox-gl-native/include/mbgl/util/work_task_impl.hpp:23
#16	0x0000000104bd86b4 in mbgl::util::RunLoop::process() at /Users/distiller/project/vendor/mapbox-gl-native/include/mbgl/util/run_loop.hpp:114
#17	0x000000018c6f8108 in CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION ()
#18	0x000000018c6f805c in __CFRunLoopDoSource0 ()
#19	0x000000018c6f77c8 in __CFRunLoopDoSources0 ()
#20	0x000000018c6f2694 in __CFRunLoopRun ()
#21	0x000000018c6f1f40 in CFRunLoopRunSpecific ()
#22	0x000000018c6f2c70 in CFRunLoopRun ()
#23	0x0000000104be6340 in mbgl::util::Threadmbgl::DefaultFileSource::Impl::Thread<std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&>(std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&, std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&)::'lambda'()::operator()() at /Users/distiller/project/vendor/mapbox-gl-native/include/mbgl/util/thread.hpp:69
#24	0x0000000104be61c4 in decltype(std::__1::forward<mbgl::util::Threadmbgl::DefaultFileSource::Impl::Thread<std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&>(std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&, std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&)::'lambda'()>(fp)()) std::__1::__invoke<mbgl::util::Threadmbgl::DefaultFileSource::Impl::Thread<std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&>(std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&, std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&)::'lambda'()>(mbgl::util::Threadmbgl::DefaultFileSource::Impl::Thread<std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&>(std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&, std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&)::'lambda'()&&) [inlined] at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:4361
#25	0x0000000104be61c0 in void std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_deletestd::__1::__thread_struct >, mbgl::util::Threadmbgl::DefaultFileSource::Impl::Thread<std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&>(std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&, std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&)::'lambda'()>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_deletestd::__1::__thread_struct >, mbgl::util::Threadmbgl::DefaultFileSource::Impl::Thread<std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&>(std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&, std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&)::'lambda'()>&, std::__1::__tuple_indices<>) [inlined] at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/thread:342
#26	0x0000000104be61bc in void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_deletestd::__1::__thread_struct >, mbgl::util::Threadmbgl::DefaultFileSource::Impl::Thread<std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&>(std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&, std::__1::shared_ptrmbgl::FileSource const&, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator > const&)::'lambda'()> >(void*) at /Applications/Xcode-11.GM.Seed.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/thread:352
```

**Comment by teddichiiwa** [comment]
I got the same error when tapping Route Overview button. The error happen randomly on iOS.

**Comment by tmpsantos** [comment]
@teddichiiwa do you have a consistent way of reproducing this issue?

**Comment by teddichiiwa** [comment]
@tmpsantos 
I don't have consistent way to reproduce it because the crash rate is low (1%).

Steps to reproduce the issue:
- Open App
- Request route from two waypoints
- Navigate using NavigationViewController
- When the NavigationViewController appear, tap on the Route Overview button
- Crashed with error: "Mapbox was compiled with optimization - stepping may behave oddly; variables may not be available."

***Platform:
    iOS

***Pods:
    pod 'Mapbox-iOS-SDK', '~> 5.5.1'
    pod 'Alamofire', '~> 4.0'
    pod 'MapboxGeocoder.swift', '~> 0.12'
    pod 'MapboxNavigation', '~> 0.39.0'
    pod 'Google-Maps-iOS-Utils', '~> 3.0.1'
    pod 'Polyline', '~> 4.0'
    pod 'SwiftyJSON', '~> 4.0'

***Xcode version: 
   11.2.1 (11B500)

***Device:
   iPhone XS (13.3.1)

**Comment by tmpsantos** [comment]
@teddichiiwa thank you for the steps! cc @julianrex

**Comment by alexshalamov** [comment]
@tmpsantos looks like an old issue that was fixed by https://github.com/mapbox/mapbox-gl-native/commit/40d42f339216e21e04b0ad2509de2fb4f6483103

**Comment by tmpsantos** [comment]
> looks like an old issue that was fixed by [40d42f3](https://github.com/mapbox/mapbox-gl-native/commit/40d42f339216e21e04b0ad2509de2fb4f6483103)

@teddichiiwa can you try with a version that inclues this fix?
